src/server/server_family.cc

```diff
@@ -2845,7 +2851,7 @@ void ServerFamily::ReplicaOfInternal(CmdArgList args, Transaction* tx, SinkReply
   }
 
   if (on_err == ActionOnConnectionFail::kReturnOnError) {
-    replica_->StartMainReplicationFiber(builder);
+    new_replica->StartMainReplicationFiber(builder);
```

Comment:
This is the FIX. Same `replica_` instance was used by two fibers and we ended up starting the main replication fiber twice. The rest of the changes above are an early exit to avoid creating the fiber all together in case of an error or if the context got cancelled.